lib/data/repository/my_anime_list_repository.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import '../model/api/request/get_top_anime_request.dart';
+import '../model/api/response/get_top_anime_response.dart';
+import '../providers/api_client_provider.dart';
+
+class MyAnimeListRepository {
+  Future<GetTopAnimeResponse> getTopAnime({
+    required Ref ref,
+    required GetTopAnimeRequest getTopAnimeRequest,
+  }) async {
+    return ref.read(apiClientProvider).getTopAnime(
+        type: getTopAnimeRequest.type,
+        page: getTopAnimeRequest.page,
+        limit: getTopAnimeRequest.limit,
+        filter: getTopAnimeRequest.filter);
```

Comment:
can we add common i guess not important but warning
filter: getTopAnimeRequest.filter, );
<img width="991" alt="image" src="https://user-images.githubusercontent.com/14831652/197664710-8aa9017e-c03b-45c8-960b-fa7c7a3d3a43.png">